CMakeLists.txt

```diff
@@ -501,6 +497,10 @@ add_custom_command(                          OUTPUT  ${syscall_macros_h}
   > ${syscall_macros_h}
   DEPENDS ${ZEPHYR_BASE}/scripts/gen_syscall_header.py
   )
+add_subdirectory(ext)
+add_subdirectory(subsys)
+add_subdirectory(drivers)
+
```

Comment:
This should be higher up so it is not in the middle of the syscall build scripts.

Reply:
This was originally placed like this, cause there has been this NCS commit where nRF was loaded earlier as module with this commit by @carlescufi 

https://github.com/pkchan/fw-nrfconnect-zephyr/commit/86599ab896febfcdf9a58f6026f49e4a45c3c103

Before re-ordering this, we should ensure we are not breaking other things.

Should modules has the possibility of defining an include priority/order ?

Reply:
> Before re-ordering this, we should ensure we are not breaking other things.

what is breaking? I tested that with almost all of ext as modules, nothing breaks.

Reply:
I have no idea why I placed that hook there @tejlmand, I don't think there was a particular reason. I know that @nashif has a reason why he moved it, and it's related to mbedtls.

Reply:
The point is moot until we have been able to reproduce the bug: https://github.com/zephyrproject-rtos/zephyr/issues/16046#issuecomment-491208500

Reply:
> 
> 
> > Before re-ordering this, we should ensure we are not breaking other things.
> 
> what is breaking? I tested that with almost all of ext as modules, nothing breaks.

I'm not saying anything is breaking, only wanted to point that in nRF @carlescufi had placed loading at that location and before moving around I wanted to be sure he didn't have a specific reason for that location, so that this PR risked breaking something else.

>  have no idea why I placed that hook there @tejlmand, I don't think there was a particular reason.

?But seems there were no specific reason.

Reply:
> what is breaking? I tested that with almost all of ext as modules, nothing breaks.

Just as additional info. Nothing would break in Zephyr.
I was concerned for downstream projects using Zephyr as module.

Reply:
> The point is moot until we have been able to reproduce the bug: [#16046 (comment)](https://github.com/zephyrproject-rtos/zephyr/issues/16046#issuecomment-491208500)

see #16065